torchx/schedulers/test/ray_scheduler_test.py

```diff
@@ -322,44 +325,81 @@ def test_list_throws_without_address(self) -> None:
                 self._scheduler.list()
```

Comment:
would be good to add some "state machine" mock tests that runs through that driver loop step by step